### DISCUSSION TITLE
Multiple/Parallel Tool calls in the AI SDK

### DISCUSSION THREAD

**Discussion by rub3nd** [vercel/community#7258] [Help]
### Summary

Is it possible to enable streaming the results of multiple tool calls at the same time?

### Example

https://chat.vercel.ai/

### Steps to Reproduce

I think the easiest way of understanding what I would like to achieve is using your guys own demo:

https://chat.vercel.ai/

My message: "what are trending memecoins today and what is the price of doge?"

I only get the trending memecoins, but not the price of doge.

**Comment by versecafe** [comment]
You can trigger parallel calls pretty simply by sending up multiple requests it streams down multiple tool call results, but triggering multiple tool calls from a single prompt doesn't work unless you make the first call then trigger the subsequent one as part of the function.

**Comment by kdawgwilk** [comment]
Is there a way to disable parallel tool calls?  We are running into issues with multiple tool calls results are being combined and end up larger than the context limit and so we don't really have a way of knowing which tool call is putting us over the limits.  cc @palmerturley34

[a comment by an automated account is not shown]
